dm/config/subtask.go

```diff
@@ -317,6 +323,29 @@ func (c *SubTaskConfig) Decode(data string, verifyDecryptPassword bool) error {
 	return c.Adjust(verifyDecryptPassword)
 }
 
+func adjustOnlineTableRules(ruleType string, rules []string) ([]string, error) {
+	adjustedRules := make([]string, 0, len(rules))
+	for _, r := range rules {
+		if !strings.HasPrefix(r, "^") {
+			r = "^" + r
```

Comment:
perhaps it would be better to just return the error here rather than adjust re ((add "^" and "$") ) for the user

Reply:
I think this way is simpler for users. They need to check the error message and rewrite task.yaml if we return an error here. Waiting for other reviewer' suggestions...

Reply:
It is ok if noted in document.